test/src/test/unit/git_protocol/upload_pack.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright © 2021 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use radicle_link_git_protocol::upload_pack;
+
+mod header {
+    use super::*;
+    use std::str::FromStr as _;
+
+    #[test]
+    fn service_must_be_upload_pack() {
+        assert_eq!(
+            upload_pack::Header::from_str("git-receive-pack "),
```

Comment:
I really like the _simple_ errors which make testing less cumbersome to write. Using structured/enum based errors anywhere might have its perks but it for sure doesn't make testing easier.